### DISCUSSION TITLE
Computing probabilities of choices

### DISCUSSION THREAD

**Discussion by deepak-facere** [sgl-project/sglang#300] [Q&A] [answered]
I was looking over the code of how `select` works. As explained in the Readme, the choice with maximum probability is selected. Looking over the code in `runtime_endpoint.py`, I came across the following snippet

```
 data = {
            "text": [s.text_ + c for c in choices],
            "sampling_params": {"max_new_tokens": 0},
            "return_logprob": True,
            "logprob_start_len": max(prompt_len - 2, 0),
        }
```

I was wondering what's the rationale behind having `"logprob_start_len": max(prompt_len - 2, 0)` Why not have this as `prompt_len`?

**Comment by merrymercy** [comment, marked as answer]
"prompt_len -2" makes it safer while not changing the results. See also https://github.com/guidance-ai/guidance/blob/main/notebooks/art_of_prompt_design/prompt_boundaries_and_token_healing.ipynb
